controllers/vspherecluster_controller.go

```diff
@@ -177,47 +177,71 @@ func (r *VSphereClusterReconciler) reconcileAPIEndpoints(ctx *context.ClusterCon
 	// Iterate over the cluster's control plane CAPI machines.
 	for _, machine := range clusterutilv1.GetControlPlaneMachines(machines) {
 
-		// Only machines with bootstrap data will have an IP address.
-		if machine.Spec.Bootstrap.Data == nil {
-			ctx.Logger.V(6).Info(
-				"skipping machine while looking for IP address",
-				"machine-name", machine.Name,
-				"skip-reason", "nilBootstrapData")
-			continue
-		}
-
-		// Get the VSphereMachine for the CAPI Machine resource.
-		vsphereMachine, err := infrautilv1.GetVSphereMachine(ctx, ctx.Client, machine.Namespace, machine.Name)
+		// Get the machine's associated KubeadmConfig resource to check if
+		// there is a ControlPlaneEndpoint value assigned to the Init/Join
+		// configuration.
+		//
+		// TODO(akutz) Assuming the Config type violates the separation model
+		//             in CAPI v1a2. Please see https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/555#issuecomment-529237211
+		//             for more information.
+		kubeadmConfig, err := infrautilv1.GetKubeadmConfigForMachine(ctx, ctx.Client, machine)
```

Comment:
This is likely to require additional RBAC permissions to work (assuming controller-runtime client probably Get, List, Watch)

Reply:
Ah, good call. I'm sure I'd find that when I do a test. Note to self, re-enable e2e on `master`.

Reply:
Done. Thanks again!